File: Presentation/NPOMS.Web/ClientApp/src/app/models/FinancialMatters.ts


export class FinancialMatters {
  constructor(
    public id: number,
    public fundingApplicationDetailId: number,
 
    public property: string,
    //public subProperty: string,
    public amountOne: number,
    public amountTwo: number,
    public amountThree: number,
    public totalFundingAmount: number,
    public createdDateTime: Date,
    public type: string) { }
}

export interface IPreviousFinancialYear {
  id: number;
  fundingApplicationDetailId: number;
  incomeDescription: string;
  incomeAmount: number;
  expenditureDescription: string;
  expenditureAmount: number;
  createdUserId: number;
  createdDateTime: Date;
  updatedUserId: number;
  updatedDateTime: Date;
}

export interface ISourceOfInformation {
  id: number;
  npoProfileId: number;
  selectedSourceValue: number;
  additionalSourceInformation: string;
  createdUserId: number;
  createdDateTime: Date;
  updatedUserId: number;
  updatedDateTime: Date;
}

export interface IAffiliatedOrganisation {
  id: number;
  fundingApplicationDetailId: number;
  nameOfOrganisation: string;
  contactPerson: string;
  emailAddress: string;
  telephoneNumber: string;
  website: string;
  createdUserId: number;
  createdDateTime: Date;
  updatedUserId: number;
  updatedDateTime: Date;
}

File: Libraries/NPOMS.Domain/Entities/FinancialMatters.cs
using NPOMS.Domain.Dropdown;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace NPOMS.Domain.Entities
{
    [Table("FinancialMatters", Schema = "fa")]
    public class FinancialMatters : BaseEntity
    {
        //public int PropertyId { get; set; }
        //public int SubPropertyId { get; set; }
        [Column(TypeName = "nvarchar(50)")]
        public string Property { get; set; }
        //[Column(TypeName = "nvarchar(50)")]
        //public string SubProperty { get; set; }
        [Column(TypeName = "numeric(18,6)")]
        public decimal AmountOne { get; set; }
        [Column(TypeName = "numeric(18,6)")]
        public decimal AmountTwo { get; set; }
        [Column(TypeName = "numeric(18,6)")]
        public decimal AmountThree { get; set; }
        public int TotalFundingAmount { get; set; }
        public int? FundingApplicationDetailId { get; set; }
        //public string Type { get; set; }
        public bool? IsActive { get; set; }
        public int? CreatedUserId { get; set; }
        public DateTime? CreatedDateTime { get; set; }
        public int? UpdatedUserId { get; set; }
        public DateTime? UpdatedDateTime { get; set; }
        public bool? ChangesRequired { get; set; }
        public bool? IsNew { get; set; }
        public FundingApplicationDetail FundingApplicationDetail { get; set; }
        //public PropertyType PropertyType { get; set; }
        //public PropertySubType PropertySubType { get; set; }
    }
}
